services/web-app/lib/github.js

```diff
@@ -153,10 +201,18 @@ export const getLibraryData = async (params = {}) => {
 
   const packageJsonContent = await getPackageJsonContent(params, library.packageJsonPath)
 
-  const filteredAssets = libraryParams.asset
+  let filteredAssets = libraryParams.asset
     ? assets.filter((asset) => getSlug(asset.content) === libraryParams.asset)
     : assets
 
+  filteredAssets = filteredAssets.filter((asset) => {
+    const isValidAsset = validateAsset(asset.content, library)
+    if (libraryParams.asset) {
+      return isValidAsset && getSlug(asset.content) === libraryParams.asset
+    }
+    return isValidAsset
+  })
```

Comment:
Do we need that first filteredAssets? I think this can be simplified to:

```suggestion
  const filteredAssets = assets.filter((asset) => {
    const isValidAsset = validateAsset(asset.content, library)
    if (libraryParams.asset) {
      return isValidAsset && getSlug(asset.content) === libraryParams.asset
    }
    return isValidAsset
  })
```

Reply:
You're right! fixed!